contracts/core/dexes/LizardDex.sol

```diff
@@ -0,0 +1,89 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.17;
+
+// imported contracts and libraries
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+// interfaces
+import "../../interface/ILiquidityDex.sol";
+import "../../interface/lizard/ILizardRouter01.sol";
+
+// libraries
+import "../../libraries/Addresses.sol";
+
+// constants and types
+import {LizardDexStorage} from "../storage/LizardDex.sol";
+
+contract LizardDex is Ownable, ILiquidityDex, LizardDexStorage {
+    using SafeERC20 for IERC20;
+
+    function doSwap(
+        uint256 _sellAmount,
+        uint256 _minBuyAmount,
+        address _receiver,
+        address[] memory _path
+    ) public override returns (uint256) {
+        address sellToken = _path[0];
+
+        IERC20(sellToken).safeIncreaseAllowance(
+            Addresses.lizardRouter,
+            _sellAmount
+        );
+
+        ILizardRouter01.Route[] memory routes = new ILizardRouter01.Route[](
+            _path.length - 1
+        );
+
+        for (uint256 idx; idx < _path.length - 1; ) {
+            address curSellToken = _path[idx];
+            address curBuyToken = _path[idx + 1];
+
+            routes[idx].from = curSellToken;
+            routes[idx].to = curBuyToken;
+            routes[idx].stable = isStable[curSellToken][curBuyToken];
+
+            unchecked {
+                ++idx;
+            }
+        }
+
+        if (!isFeeOnTransfer[_path[0]][_path[_path.length - 1]]) {
+            ILizardRouter01(Addresses.lizardRouter).swapExactTokensForTokens(
+                _sellAmount,
+                _minBuyAmount,
+                routes,
+                _receiver,
+                block.timestamp
+            );
+        } else {
+            ILizardRouter01(Addresses.lizardRouter)
+                .swapExactTokensForTokensSupportingFeeOnTransferTokens(
```

Comment:
I think we can generalise to this call if we're concerned about tokens with transfer fees. It should work perfectly fine for tokens without a transfer fee too, if I'm not mistaken. So in that case we do not need to store `isFeeOnTransfer` and check for it.

Reply:
Yes, let me test the gas difference, and we can decide if we just simplified it.